llama-index-integrations/vector_stores/llama-index-vector-stores-mariadb/llama_index/vector_stores/mariadb/base.py

```diff
@@ -0,0 +1,445 @@
+            return "="
+
+    def _build_filter_clause(self, filter_: MetadataFilter) -> str:
+        filter_value = filter_.value
+        if filter_.operator in [FilterOperator.IN, FilterOperator.NIN]:
+            values = []
+            for v in filter_.value:
+                if isinstance(v, str):
+                    value = f"'{v}'"
+
+                values.append(value)
+            filter_value = ", ".join(values)
+            filter_value = f"({filter_value})"
+        elif isinstance(filter_.value, str):
+            filter_value = f"'{filter_.value}'"
+
+        return f"JSON_VALUE(metadata, '$.{filter_.key}') {self._to_mariadb_operator(filter_.operator)} {filter_value}"
+
+    def _filters_to_where_clause(self, filters: MetadataFilters) -> str:
+        conditions = {
+            FilterCondition.OR: "OR",
+            FilterCondition.AND: "AND",
+        }
+        if filters.condition not in conditions:
+            raise ValueError(
+                f"Unsupported condition: {filters.condition}. "
+                f"Must be one of {list(conditions.keys())}"
+            )
+
+        clauses: List[str] = []
+        for filter_ in filters.filters:
+            if isinstance(filter_, MetadataFilter):
+                clauses.append(self._build_filter_clause(filter_))
+                continue
+
+            if isinstance(filter_, MetadataFilters):
+                subfilters = self._filters_to_where_clause(filter_)
+                if subfilters:
+                    clauses.append(f"({subfilters})")
+                continue
+
+            raise ValueError(
+                f"Unsupported filter type: {type(filter_)}. Must be one of {MetadataFilter}, {MetadataFilters}"
+            )
+        return f" {conditions[filters.condition]} ".join(clauses)
+
+    def _db_rows_to_query_result(
+        self, rows: List[DBEmbeddingRow]
+    ) -> VectorStoreQueryResult:
+        nodes = []
+        similarities = []
+        ids = []
+        for db_embedding_row in rows:
+            node = metadata_dict_to_node(db_embedding_row.metadata)
+            node.set_content(str(db_embedding_row.text))
+
+            similarities.append(db_embedding_row.similarity)
+            ids.append(db_embedding_row.node_id)
+            nodes.append(node)
+
+        return VectorStoreQueryResult(
+            nodes=nodes,
+            similarities=similarities,
+            ids=ids,
+        )
+
+    def query(self, query: VectorStoreQuery, **kwargs: Any) -> VectorStoreQueryResult:
+        if query.mode != VectorStoreQueryMode.DEFAULT:
+            raise NotImplementedError(f"Query mode {query.mode} not available.")
+
+        self._initialize()
+
+        stmt = f"""
+        SELECT
+            node_id,
+            text,
+            embedding,
+            metadata,
+            vec_distance(embedding, vec_fromtext('{query.query_embedding}')) AS distance
+        FROM {self.table_name}
```

Comment:
Can we make table_name a sql variable here too?

Reply:
Please, check my comment above.